### PR TITLE
ci: Separate tests to run them in parallel

### PR DESCRIPTION
**Pull request by Itxaka**
Instead of running the test all at the same time, we can
separate them using a matrix so they run in parallel

This patch also separates the upgrade tests into using
signed and unsigned images so we can also run those in
parallel, as they are the ones taking most of the time

Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

**Review by mudler — APPROVED** [review]
Looks good here :+1:

File: tests/upgrades-images-signed/tests_suite_test.go [hunk 1]
```diff
@@ -9,5 +9,5 @@ import (
 
 func TestTests(t *testing.T) {
 	RegisterFailHandler(Fail)
-	RunSpecs(t, "cOS Upgrate test Suite")
+	RunSpecs(t, "cOS Upgrate test Suite - Images signed")
```
Comment (mudler) [inline tests/upgrades-images-signed/tests_suite_test.go:12]:
```suggestion
	RunSpecs(t, "cOS Upgrate test Suite - Signed images")
```

File: tests/upgrades-images-signed/upgrade_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package cos_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/rancher-sandbox/cOS/tests/sut"
+)
+
+var _ = Describe("cOS Upgrade tests - Images signed", func() {
```
Comment (mudler) [inline tests/upgrades-images-signed/upgrade_test.go:9]:
```suggestion
var _ = Describe("cOS Upgrade tests - Signed images", func() {
```

**Comment by Itxaka** [comment]
I'll change the names in a post commit as to not lose the approval :)

**Comment by mudler** [comment]
No worries :) I think we don't dismiss approvals at the moment :+1:

**Pull request merged**